### PR TITLE
Upgrade Thrift to 0.10.0, bump Pucket to 1.5.0.

### PR DESCRIPTION
**Pull request by nathankleyn**
Upgrade Thrift to the latest version, bump the major version of this repository as it'll be a breaking change for some.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+
+# Scala-IDE specific
+.ensime
+.ensime_cache/
+.scala_dependencies
+.worksheet
+
+# ENSIME specific
+.ensime_cache/
+.ensime
```
Comment (albertpastrana) [inline .gitignore]: these two ignores are declared twice :)

Comment (nathankleyn) [reply .gitignore]: Oops! Teach me to copy and paste this from the gitignore repository! I'll fix.

Comment (nathankleyn) [reply .gitignore]: Fixed. 👍

**Review by albertpastrana — APPROVED** [review]
LGTM

**Pull request merged**